components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle

```diff
@@ -1,64 +1,159 @@
 import org.gradle.util.GradleVersion
 import java.nio.charset.StandardCharsets;
 
+if (GradleVersion.current() < GradleVersion.version("5.0")) {
+    throw new IllegalStateException("Build validation not supported for Gradle ${GradleVersion.current()}. Upgrade your project's build to Gradle 5 or newer.")
+}
+
+// note that there is no mechanism to share code between the initscript{} block and the main script, so some logic is duplicated
+
+// conditionally apply the GE / Build Scan plugin to the classpath so it can be applied to the build further down in this script
 initscript {
-    def gradleEnterprisePluginVersion = "3.10.3"
+    def isTopLevelBuild = !gradle.parent
+    if (!isTopLevelBuild) {
+        return
+    }
+
+    def enableGradleEnterprise = Boolean.getBoolean("com.gradle.enterprise.build_validation.enableGradleEnterprise")
+
+    def gePluginVersion = enableGradleEnterprise ? "3.10.3" : null
+    def ccudPluginVersion = enableGradleEnterprise ? "1.7.2" : null
 
     repositories {
         gradlePluginPortal()
     }
 
     dependencies {
-        classpath("com.gradle:gradle-enterprise-gradle-plugin:${gradleEnterprisePluginVersion}")
+        if (gePluginVersion) {
+            classpath "com.gradle:gradle-enterprise-gradle-plugin:$gePluginVersion"
+        }
+
+        if (ccudPluginVersion) {
+            classpath "com.gradle:common-custom-user-data-gradle-plugin:$ccudPluginVersion"
+        }
+    }
+}
+
+def BUILD_SCAN_PLUGIN_ID = 'com.gradle.build-scan'
+def BUILD_SCAN_PLUGIN_CLASS = 'com.gradle.scan.plugin.BuildScanPlugin'
+
+def GRADLE_ENTERPRISE_PLUGIN_ID = 'com.gradle.enterprise'
+def GRADLE_ENTERPRISE_PLUGIN_CLASS = 'com.gradle.enterprise.gradleplugin.GradleEnterprisePlugin'
+def GRADLE_ENTERPRISE_EXTENSION_CLASS = 'com.gradle.enterprise.gradleplugin.GradleEnterpriseExtension'
+
+def CCUD_PLUGIN_ID = 'com.gradle.common-custom-user-data-gradle-plugin'
+def CCUD_PLUGIN_CLASS = 'com.gradle.CommonCustomUserDataGradlePlugin'
+
+def isTopLevelBuild = !gradle.parent
+if (!isTopLevelBuild) {
+    return
+}
+
+def enableGradleEnterprise = Boolean.getBoolean("com.gradle.enterprise.build_validation.enableGradleEnterprise")
+
+clearWarnings()
+
+// define a buildScanPublished listener that captures the build scan URL and sends it in a message to the server
```

Comment:
-1: Looks like an inaccurate comment.

Reply:
Good point. This was copied from the "Gold standard" script and should be modified as appropriate